features/eolearn/tests/test_feature_manipulation.py

```diff
@@ -43,6 +43,52 @@ def test_content_after_timefilter(self):
         self.assertEqual(new_interval, updated_interval)
         self.assertEqual(new_timestamps, updated_timestamps)
 
+    def test_value_fillout(self):
+        feature = (FeatureType.DATA, 'TEST')
+        shape = (8, 10, 10, 5)
+        data = np.random.randint(0, 100, size=shape).astype(np.float)
+        eopatch = EOPatch(data={'TEST': data})
+
+        self.assertRaises(ValueError, ValueFilloutTask, feature, operations='x')
+        self.assertRaises(ValueError, ValueFilloutTask, feature, operations=4)
+
+        self.assertRaises(ValueError, ValueFilloutTask.fill, None, operation='f')
+        self.assertRaises(ValueError, ValueFilloutTask.fill, np.zeros((4, 5)), operation='x')
+
+        # nothing to be filled, return the same eopatch object immediately
+        eopatch_new = ValueFilloutTask(feature, operations='fb', axis=0)(eopatch)
+        self.assertEqual(eopatch, eopatch_new)
+
+        eopatch[feature][0, 0, 0, :] = np.nan
+
+        # filling forward temporally should not fill nans
+        eopatch = ValueFilloutTask(feature, operations='f', axis=0)(eopatch)
+        self.assertTrue(np.isnan(eopatch[feature][0, 0, 0, :]).all())
+
+        # filling in any direction along axis=-1 should also not fill nans since all neighbors are nans
+        eopatch = ValueFilloutTask(feature, operations='fb', axis=-1)(eopatch)
+        self.assertTrue(np.isnan(eopatch[feature][0, 0, 0, :]).all())
+
+        # filling nans backwards temporally should fill nans
+        eopatch = ValueFilloutTask(feature, operations='b', axis=0)(eopatch)
+        self.assertFalse(np.isnan(eopatch[feature]).any())
+
+        # try filling something else than nan (e.g.: -1)
+        eopatch[feature][0, :, 0, 0] = -1
+        eopatch = ValueFilloutTask(feature, operations='b', value=-1, axis=-1)(eopatch)
+        # !!
+        self.assertFalse(np.isnan(eopatch[feature]).any())
```

Comment:
Did you want to check here that there are no -1 values? This check for nan values doesn't hurt, but the main thing here should be to check if all -1 values have been replaced.